src/hooks/useListing.js

```diff
@@ -0,0 +1,32 @@
+import { useQuery } from 'react-query';
+import axios from 'axios';
+import {apiEndpoint} from "../constants";
+
+const getListings = async ( filters ) => {
+
+  const params = new URLSearchParams({
+    cardOrigin: filters.cardType,
+    saleToken: filters.token,
+    cardRarity: filters.rarity,
+    search: filters.keyword,
+    sortBy: filters.sortField
+  })
+
+  const response = await axios.get(`${apiEndpoint}/listing/${params.toString()}`);
+
+  if (response.status !== 200) {
+      console.error(response.statusText);
+      throw new Error("Failed to get profile.");
+  }
+
+  return response.data;
+};
+
+export const useFetchListingQuery = ( filters ) =>
+  useQuery({
```

Comment:
Can we use infiniteQuery here? So it can support infinite scroll/paging out of the box. The paging on server is done now, you can pass in `offset` and `limit` to restrict the # of listings, and it will return a `nextOffset` to tell you the next page.

Reply:
I've fixed this.